### PR TITLE
Added JsonForecast datasource

### PR DESCRIPTION
**Pull request by in4margaret**
# Pull Request

Issue Number: [(Link to Github Issue or Azure Dev Ops Task/Story)](https://github.com/microsoft/carbon-aware-sdk/issues/258)

## Summary

Added JsonForecast datasource

## Changes

- [x]  JsonDataSource.cs implements IForecastDataSource. 
- [x] AzureRegionTestDataGenerator.cs generates data compatible with IForecastDataSource  implementation for JsonDataSource
- [x] Json Data Source Mocks generates data needed for Forecast Integration tests.
- [x] Forecast Endpoints are enabled for Json
- [x] JSON unit tests pass
- [x] Integration tests with Json pass for Emissions & Forecast endpoints
- [x] Documentation is updated
- [x] Remove Newtonsoft package from CarbonAware.Tools.AzureRegionTestDataGenerator and use System.Text.Json instead
- [x] Changed ServiceCollectionExtensions from JSON datasource to match signature of other datasources (return IServiceCollection instead of void). 

## Checklist

- [ ] Local Tests Passing?
- [ ] CICD and Pipeline Tests Passing?
- [ ] Added any new Tests?
- [ ] Documentation Updates Made?
- [ ] Are there any API Changes? If yes, please describe below.
- [ ] This is not a breaking change. If it is, please describe it below.

## Are there API Changes?

If yes, what are the expected API Changes? Please link to an API-Comparison
workflow with the API Diff.

## Is this a breaking change?

If yes, what workflow does this break?

## Anything else?

Other comments, collaborators, etc.

> Please follow
> [GitHub's suggested syntax](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
> to link Pull Requests to Issues via keywords

This PR Closes #<issue_number>

### CODE REVIEW

File: src/CarbonAware.WebApi/test/integrationTests/IntegrationTestingBase.cs [hunk 1]
```diff
@@ -127,6 +127,15 @@ public void Setup()
     [SetUp]
     public void SetupTests()
     {
+        if (_dataSource == DataSourceType.JSON)
+        {
+            // To force WebApplication to consume new JSON datasorce it needs to be restarted.
```
Comment (Kpopovs1) [inline src/CarbonAware.WebApi/test/integrationTests/IntegrationTestingBase.cs:132]: spelling nits: `datasource` and `CarbonAware`

**Review by Kpopovs1 — COMMENTED** [review]
Moving from Newtonsoft to System.Text.Json was a good catch! I just added few comments.

File: src/CarbonAware.WebApi/test/integrationTests/IntegrationTestingBase.cs (hunk 1 above)
Comment (in4margaret) [reply src/CarbonAware.WebApi/test/integrationTests/IntegrationTestingBase.cs]: fixed

**Review by vaughanknight — APPROVED** [review]
These changes look great.  Awesome gap to fill to so thank you.  Very much appreciated.

**Comment by vaughanknight** [comment]
From the errors it looks like the test data json file might have issues, I need to look into it.  I don't think it's a code issue at this time, but likely a test data file issue.

**Comment by Willmish** [comment]
@vaughanknight could you confirm whether we are holding off with merging this until after the release?

**Comment by Willmish** [comment]
bringing this back up, to confirm whether fixing this is being held off until post v1.1 @vaughanknight

**Comment by Willmish** [comment]
From #405 : currently available to pick up for review

**Comment by Willmish** [comment]
#417 To decide whether to close this off on next meeting - has been broken for a while and no one volunteered to fix/pick this up.

**Comment by danuw** [comment]
This has been breaking for too long and is not a critical feature. It was a great contribution we will revisit using the latest code. 
We will be abandoning this version for now.

**Pull request closed without merging**